lib/config.py

```diff
@@ -294,14 +295,22 @@ def parse(self, command_line_args=None):
             for key, value in command_line_args.items():
                 if key in config[node_name]:
                     if value is not None:
-                        config[node_name][key] = value
+                        value_type = type(config[node_name][key])
+                        if  value_type == str or value_type == bool or value_type == list:
+                            config[node_name][key] = value
+                        elif type(config[node_name][key]) == dict:
+                            config[node_name][key] = json.loads(value)
                     command_line_args.pop(key)
         # update common config node with remaining command line arguments
         node_name = "common"
         for key, value in command_line_args.items():
             if key in config[node_name]:
                 if value is not None:
-                    config[node_name][key] = value
+                    value_type = type(config[node_name][key])
+                    if value_type == str or value_type == bool or value_type == list:
+                        config[node_name][key] = value
+                    elif type(config[node_name][key]) == dict:
+                        config[node_name][key] = json.loads(value)
                 command_line_args.pop(key)
```

Comment:
How integer values, like `log_size`, would be handled?

What if we abstracted this assignment logic to avoid code duplication?  For example:

```python
def assign_config_value(name, value):
  value_type = type(name)
  if value_type == dict:
    name = json.loads(value)
  else:
    name = value
```

Then `assign_config_value` could be used:

```python
for key, value in command_line_args.items():
  if key in config[node_name]:
    if value is not None:
      assign_config_value(config[node_name][key], value)
    command_line_args.pop(key)
```

@olavphibm @fabianorosas What do you think?